core/server/worker/src/main/java/alluxio/worker/page/BlockPageId.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.page;
+
+import alluxio.client.file.cache.PageId;
+
+import com.google.common.base.Preconditions;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Specialized {@link PageId} when it's part of a block.
+ */
+public final class BlockPageId extends PageId {
+  // the name pattern of the page files of a block
+  // the block id and size are encoded as 16-byte hexadecimal
+  private static final Pattern FILE_ID_PATTERN =
+      Pattern.compile("paged_block_([0-9a-fA-F]{16})_size_([0-9a-fA-F]{16})");
+  private static final String FILE_ID_TEMPLATE = "paged_block_%016x_size_%016x";
```

Comment:
I would use a simpler template %016x_%016x to reduce the parsing cost, your call.

Reply:
My concern is that if the user somehow misconfigures client local cache and worker page store to share the same cache directory, the special naming scheme is the only thing that may help prevent we messing up the pages, as other apps may pretty well also use a naming scheme as simple as `number_number`.